test/Interop/Cxx/operators/Inputs/member-inline.h

```diff
@@ -477,4 +477,24 @@ struct HasOperatorCallWithDefaultArg {
   int operator()(int x = 0) const { return value + x; }
 };
 
+struct ClassWithOperatorStarAvailable {
+
+  mutable int value;
+
+  int *operator->() const { return &value; }
+
+  int &operator*() const { return *operator->(); }
+};
+
+struct ClassWithOperatorStarUnavailable {
+
+  mutable int value;
+
+  int *operator->() const { return &value; }
+
+  int &operator*() const __attribute__((availability(swift, unavailable))) {
+    return *operator->();
+  }
+};
+
```

Comment:
Could you please add a test that checks that inherited `operator*` gets the correct attributes, e.g. `struct DerivedHasOperatorStar : HasOperatorStar`

Reply:
This case is not covered in this patch. Added a test case for this quirk with comment that a later patch would fix the case where the `operator *` is in base class and needs to be hidden in derived class also.